Add modelpicker and ollama support

BalanceBalls opened the pull request:
Great project! Just what I was looking for

**The changes introduced**
- a model picker instead of input field (via /v1/models endpoint)
- out-of-the-box support for local models and other OpenAI API compatible providers
  * ability to specify only base url without endpoint path (old format is also compatible)
  * ability to specify default model (since other providers won't have gpt-3.5-turbo)
- minor code cleanup and refactoring

**ModelPicker demo**

https://github.com/user-attachments/assets/e5ce4a99-ffe0-4e54-a514-bf50cf767bc0

tearingItUp786 commented:
Oh wow! Thank you for your PR 😄 . 

I believe we need to change `OPENAI_API_KEY` to something more general. `API_KEY` sounds a little too vague so I'll leave it up to you!

tearingItUp786 reviewed (changes requested):
Hey friend. Great PR. I have a few suggestions regarding visual styling but one dealbreaker is that I notice that upon updating the model, it is not persisted in the database. 

I presume it's because of these lines of code...


```go
		inputValue := m.textInput.Value()
		if inputValue == "" {
			return cmd
		}

		switch m.mode {
		case modelMode:
			m.settings.Model = inputValue

```

We are not longer updating the m.textInput but rather are using a list so we'll have to tweak this code accordingly.

BalanceBalls commented on config/config.json:
```diff
@@ -1,4 +1,5 @@
 {
   "chatGPTApiUrl": "https://api.openai.com/v1/chat/completions",
```
Thanks for review! Since it is backwards compatible, I think it's fine to leave it as it is right now. I agree that it makes sense to change it, but there was already a lot of changes in this PR, so I would rather rename it in a different PR

BalanceBalls commented on components/models-list.go:
```diff
@@ -0,0 +1,80 @@
+	Foreground(lipgloss.Color("#fff"))
+
+var listItemSpanSelected = lipgloss.NewStyle().
+	Foreground(lipgloss.Color("#fffaaa"))
```
done

BalanceBalls commented on components/models-list.go:
```diff
@@ -0,0 +1,80 @@
+	quitting bool
+}
+
+var listItemSpan = lipgloss.NewStyle().
```
done

BalanceBalls commented:
> Hey friend. Great PR. I have a few suggestions regarding visual styling but one dealbreaker is that I notice that upon updating the model, it is not persisted in the database.
> 
> I presume it's because of these lines of code...
> 
> ```go
> 		inputValue := m.textInput.Value()
> 		if inputValue == "" {
> 			return cmd
> 		}
> 
> 		switch m.mode {
> 		case modelMode:
> 			m.settings.Model = inputValue
> ```
> 
> We are not longer updating the m.textInput but rather are using a list so we'll have to tweak this code accordingly.

Indeed. Fixed that, thanks. Also fixed ESC key handling for the model picker. Should be good now. The code is kinda messy for now though, but I plan to do components refactoring later to improve maintainability and will tidy up settings management as well :)

tearingItUp786 commented:
> > Hey friend. Great PR. I have a few suggestions regarding visual styling but one dealbreaker is that I notice that upon updating the model, it is not persisted in the database.
> > I presume it's because of these lines of code...
> > ```go
> > 		inputValue := m.textInput.Value()
> > 		if inputValue == "" {
> > 			return cmd
> > 		}
> > 
> > 		switch m.mode {
> > 		case modelMode:
> > 			m.settings.Model = inputValue
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We are not longer updating the m.textInput but rather are using a list so we'll have to tweak this code accordingly.
> 
> Indeed. Fixed that, thanks. Also fixed ESC key handling for the model picker. Should be good now. The code is kinda messy for now though, but I plan to do components refactoring later to improve maintainability and will tidy up settings management as well :)

Will pull tomorrow and try it out. Getting over the flu right now!
